Re-implement recent activity

notrodes opened the pull request:
Closes #180. Its 60% working, things that don't work:

- Colors
- max scores and percentage
- clicking on rows

Commit:
Begin reimplementing recentActivity. Uncommenting lots of code, changed the name of some properties of an object because scraper returns different names.

psvenk commented on src/scrape.ts:
```diff
@@ -360,7 +360,7 @@ async function get_recent(session: Session): Promise<Recent> {
     [...document.querySelectorAll("gradebookScore")].map(x =>
       Object.fromEntries([
-        "date", "classname", "score", "assignment",
+        "date", "classname", "grade", "assignmentname",
```
This change, along with the associated change on the frontend, causes unneeded API breakage. The names might not be ideal, but it's better just to leave them as is, so that we can avoid incompatibilities.

psvenk commented on public/js/home.js:
```diff
@@ -185,8 +186,8 @@ let recentActivity = new Tabulator("#recentActivity", {
+        {title: "Score", field: "grade", formatter: rowFormatter, headerSort: false},
         {title: "Max Score", field: "max_score", formatter: rowFormatter, headerSort: false},
         {title: "Percentage", field: "percentage", formatter: rowGradeFormatter},
     ],
```
See comment on scraper.

psvenk commented on public/js/home.js:
```diff
@@ -274,8 +275,8 @@ let mostRecentTable = new Tabulator("#mostRecentTable", {
+        {title: "Score", field: "grade", formatter: rowFormatter, headerSort: false},
         {title: "Max Score", field: "max_score", formatter: rowFormatter, headerSort: false},
         {title: "Percentage", field: "percentage", formatter: rowGradeFormatter},
     ],
```
See comment on scraper.

notrodes commented:
They are necessary changes because Aspen changed the names of the where the data is stored.

psvenk committed:
Change field names back for recent activity

To avoid API breakage, rename Aspen's fields to our fields instead of
adapting ourselves to Aspen's changes.

psvenk reviewed (dismissed):
Looks mostly good to me. The only thing left to do here is to make sure that the max score and percent for recent assignments are filled out properly. We could do this by looking up the assignment in the grades table using the class name and assignment name.

jshaw-7 commented:
not sure if this is purposeful, but the recent activity only shows up when you first load the grades page. if you click on a specific class and then hit the reset button, the recent activity is gone. also not sure how i feel about the color scheme, colors based on the grades might be better than doing it by class.
